**CRYPTOCURRENCY**

* Cryptocurrency is a digital payment system. It's a peer-to-peer system that can enable anyone from anywhere to send and receive payments. When you transfer cryptocurrency funds, the transactions are recorded in a public ledger. Cryptocurrency is stored in digital wallets.

* A cryptocurrency is an encrypted data string that denotes a unit of currency. It is monitored and organized by a peer-to-peer network called a blockchain, which also serves as a secure ledger of transactions, e.g., buying, selling, and transferring. Bitcoin, Ether, Litecoin, and Monero are popular cryptocurrencies.

**IMPORTING LIBRARIES**

In [61]:
import pandas as pd
import numpy as np

from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split

import matplotlib.pyplot as plt
from itertools import cycle
import plotly.graph_objects as go
import plotly.express as px
from plotly.subplots import make_subplots

**LOADING THE DATA**

**Binance Coin**

BNB Coin is a cryptocurrency that is used primarily to pay transaction and trading fees on the Binance exchange.

In [62]:
Binance_coin = pd.read_csv("coin_BinanceCoin.csv")
Binance_coin

SNo          Name Symbol                 Date        High         Low  \
0        1  Binance Coin    BNB  2017-07-26 23:59:59    0.109013    0.099266   
1        2  Binance Coin    BNB  2017-07-27 23:59:59    0.108479    0.100888   
2        3  Binance Coin    BNB  2017-07-28 23:59:59    0.109019    0.101473   
3        4  Binance Coin    BNB  2017-07-29 23:59:59    0.111264    0.101108   
4        5  Binance Coin    BNB  2017-07-30 23:59:59    0.108138    0.103162   
...    ...           ...    ...                  ...         ...         ...   
1437  1438  Binance Coin    BNB  2021-07-02 23:59:59  290.621683  277.350302   
1438  1439  Binance Coin    BNB  2021-07-03 23:59:59  302.605867  283.434010   
1439  1440  Binance Coin    BNB  2021-07-04 23:59:59  314.713019  292.787370   
1440  1441  Binance Coin    BNB  2021-07-05 23:59:59  307.684855  293.523465   
1441  1442  Binance Coin    BNB  2021-07-06 23:59:59  321.520965  302.195584   

            Open       Close        Volume     Marketcap  
0       0.105893    0.105138  2.003950e+05  1.051380e+07  
1       0.105108    0.107737  3.444990e+05  1.077370e+07  
2       0.107632    0.104067  3.425680e+05  1.040670e+07  
3       0.104782    0.107811  3.402180e+05  1.078110e+07  
4       0.107935    0.106414  2.242610e+05  1.064140e+07  
...          ...         ...           ...           ...  
1437  287.754456  287.423094  1.133633e+09  4.410016e+10  
1438  287.215592  298.237117  1.113777e+09  4.575938e+10  
1439  298.113557  307.732096  1.387396e+09  4.721623e+10  
1440  307.684855  302.377980  1.504870e+09  4.639473e+10  
1441  302.195584  320.934802  2.203265e+09  4.924196e+10  

[1442 rows x 10 columns]

**ETHEREUM**

Ethereum is a platform powered by blockchain technology that is best known for its native cryptocurrency, called ether, or ETH, or simply ethereum. The distributed nature of blockchain technology is what makes the Ethereum platform secure, and that security enables ETH to accrue value.

In [63]:
Ethereum = pd.read_csv("coin_Ethereum.csv")
Ethereum

SNo      Name Symbol                 Date         High          Low  \
0        1  Ethereum    ETH  2015-08-08 23:59:59     2.798810     0.714725   
1        2  Ethereum    ETH  2015-08-09 23:59:59     0.879810     0.629191   
2        3  Ethereum    ETH  2015-08-10 23:59:59     0.729854     0.636546   
3        4  Ethereum    ETH  2015-08-11 23:59:59     1.131410     0.663235   
4        5  Ethereum    ETH  2015-08-12 23:59:59     1.289940     0.883608   
...    ...       ...    ...                  ...          ...          ...   
2155  2156  Ethereum    ETH  2021-07-02 23:59:59  2155.596496  2021.824808   
2156  2157  Ethereum    ETH  2021-07-03 23:59:59  2237.567155  2117.590013   
2157  2158  Ethereum    ETH  2021-07-04 23:59:59  2384.286857  2190.837703   
2158  2159  Ethereum    ETH  2021-07-05 23:59:59  2321.922836  2163.041394   
2159  2160  Ethereum    ETH  2021-07-06 23:59:59  2346.294874  2197.919385   

             Open        Close        Volume     Marketcap  
0        2.793760     0.753325  6.741880e+05  4.548689e+07  
1        0.706136     0.701897  5.321700e+05  4.239957e+07  
2        0.713989     0.708448  4.052830e+05  4.281836e+07  
3        0.708087     1.067860  1.463100e+06  6.456929e+07  
4        1.058750     1.217440  2.150620e+06  7.364501e+07  
...           ...          ...           ...           ...  
2155  2109.892677  2150.040364  3.179621e+10  2.505527e+11  
2156  2150.835025  2226.114282  1.743336e+10  2.594475e+11  
2157  2226.550382  2321.724112  1.878711e+10  2.706217e+11  
2158  2321.922836  2198.582464  2.010379e+10  2.562978e+11  
2159  2197.919385  2324.679449  2.089186e+10  2.710286e+11  

[2160 rows x 10 columns]

**BITCOIN**

Bitcoin is a digital currency which operates free of any central control or the oversight of banks or governments. Instead it relies on peer-to-peer software and cryptography. A public ledger records all bitcoin transactions and copies are held on servers around the world.

In [64]:
Bitcoin = pd.read_csv("coin_Bitcoin.csv")
Bitcoin

SNo     Name Symbol                 Date          High           Low  \
0        1  Bitcoin    BTC  2013-04-29 23:59:59    147.488007    134.000000   
1        2  Bitcoin    BTC  2013-04-30 23:59:59    146.929993    134.050003   
2        3  Bitcoin    BTC  2013-05-01 23:59:59    139.889999    107.720001   
3        4  Bitcoin    BTC  2013-05-02 23:59:59    125.599998     92.281898   
4        5  Bitcoin    BTC  2013-05-03 23:59:59    108.127998     79.099998   
...    ...      ...    ...                  ...           ...           ...   
2986  2987  Bitcoin    BTC  2021-07-02 23:59:59  33939.588699  32770.680780   
2987  2988  Bitcoin    BTC  2021-07-03 23:59:59  34909.259899  33402.696536   
2988  2989  Bitcoin    BTC  2021-07-04 23:59:59  35937.567147  34396.477458   
2989  2990  Bitcoin    BTC  2021-07-05 23:59:59  35284.344430  33213.661034   
2990  2991  Bitcoin    BTC  2021-07-06 23:59:59  35038.536363  33599.916169   

              Open         Close        Volume     Marketcap  
0       134.444000    144.539993  0.000000e+00  1.603769e+09  
1       144.000000    139.000000  0.000000e+00  1.542813e+09  
2       139.000000    116.989998  0.000000e+00  1.298955e+09  
3       116.379997    105.209999  0.000000e+00  1.168517e+09  
4       106.250000     97.750000  0.000000e+00  1.085995e+09  
...            ...           ...           ...           ...  
2986  33549.600177  33897.048590  3.872897e+10  6.354508e+11  
2987  33854.421362  34668.548402  2.438396e+10  6.499397e+11  
2988  34665.564866  35287.779766  2.492431e+10  6.615748e+11  
2989  35284.344430  33746.002456  2.672155e+10  6.326962e+11  
2990  33723.509655  34235.193451  2.650126e+10  6.418992e+11  

[2991 rows x 10 columns]

**CREATION OF A TARGET COLUMN**

* Creation of a variable for predicting 'N' days out into the future(here, let us assume N = 5).

* Creation of a new column with a name Prediction.





In [65]:
################################################### Monero #############################################################
#Creation of a variable for predicting '5' days out into the future
projection_Binance_coin = 5
#creation of a new column with a name prediction
Binance_coin['Prediction'] = Binance_coin[['Close']].shift(-projection_Binance_coin)
Binance_coin

SNo          Name Symbol                 Date        High         Low  \
0        1  Binance Coin    BNB  2017-07-26 23:59:59    0.109013    0.099266   
1        2  Binance Coin    BNB  2017-07-27 23:59:59    0.108479    0.100888   
2        3  Binance Coin    BNB  2017-07-28 23:59:59    0.109019    0.101473   
3        4  Binance Coin    BNB  2017-07-29 23:59:59    0.111264    0.101108   
4        5  Binance Coin    BNB  2017-07-30 23:59:59    0.108138    0.103162   
...    ...           ...    ...                  ...         ...         ...   
1437  1438  Binance Coin    BNB  2021-07-02 23:59:59  290.621683  277.350302   
1438  1439  Binance Coin    BNB  2021-07-03 23:59:59  302.605867  283.434010   
1439  1440  Binance Coin    BNB  2021-07-04 23:59:59  314.713019  292.787370   
1440  1441  Binance Coin    BNB  2021-07-05 23:59:59  307.684855  293.523465   
1441  1442  Binance Coin    BNB  2021-07-06 23:59:59  321.520965  302.195584   

            Open       Close        Volume     Marketcap  Prediction  
0       0.105893    0.105138  2.003950e+05  1.051380e+07    0.104250  
1       0.105108    0.107737  3.444990e+05  1.077370e+07    0.099867  
2       0.107632    0.104067  3.425680e+05  1.040670e+07    0.100452  
3       0.104782    0.107811  3.402180e+05  1.078110e+07    0.106642  
4       0.107935    0.106414  2.242610e+05  1.064140e+07    0.107494  
...          ...         ...           ...           ...         ...  
1437  287.754456  287.423094  1.133633e+09  4.410016e+10         NaN  
1438  287.215592  298.237117  1.113777e+09  4.575938e+10         NaN  
1439  298.113557  307.732096  1.387396e+09  4.721623e+10         NaN  
1440  307.684855  302.377980  1.504870e+09  4.639473e+10         NaN  
1441  302.195584  320.934802  2.203265e+09  4.924196e+10         NaN  

[1442 rows x 11 columns]

In [66]:
################################################### Ethereum #############################################################
#Creation of a variable for predicting '5' days out into the future
projection_Ethereum = 5
#creation of a new column with a name prediction
Ethereum['Prediction'] = Ethereum[['Close']].shift(-projection_Ethereum)
Ethereum

SNo      Name Symbol                 Date         High          Low  \
0        1  Ethereum    ETH  2015-08-08 23:59:59     2.798810     0.714725   
1        2  Ethereum    ETH  2015-08-09 23:59:59     0.879810     0.629191   
2        3  Ethereum    ETH  2015-08-10 23:59:59     0.729854     0.636546   
3        4  Ethereum    ETH  2015-08-11 23:59:59     1.131410     0.663235   
4        5  Ethereum    ETH  2015-08-12 23:59:59     1.289940     0.883608   
...    ...       ...    ...                  ...          ...          ...   
2155  2156  Ethereum    ETH  2021-07-02 23:59:59  2155.596496  2021.824808   
2156  2157  Ethereum    ETH  2021-07-03 23:59:59  2237.567155  2117.590013   
2157  2158  Ethereum    ETH  2021-07-04 23:59:59  2384.286857  2190.837703   
2158  2159  Ethereum    ETH  2021-07-05 23:59:59  2321.922836  2163.041394   
2159  2160  Ethereum    ETH  2021-07-06 23:59:59  2346.294874  2197.919385   

             Open        Close        Volume     Marketcap  Prediction  
0        2.793760     0.753325  6.741880e+05  4.548689e+07     1.82767  
1        0.706136     0.701897  5.321700e+05  4.239957e+07     1.82787  
2        0.713989     0.708448  4.052830e+05  4.281836e+07     1.68890  
3        0.708087     1.067860  1.463100e+06  6.456929e+07     1.56603  
4        1.058750     1.217440  2.150620e+06  7.364501e+07     1.20361  
...           ...          ...           ...           ...         ...  
2155  2109.892677  2150.040364  3.179621e+10  2.505527e+11         NaN  
2156  2150.835025  2226.114282  1.743336e+10  2.594475e+11         NaN  
2157  2226.550382  2321.724112  1.878711e+10  2.706217e+11         NaN  
2158  2321.922836  2198.582464  2.010379e+10  2.562978e+11         NaN  
2159  2197.919385  2324.679449  2.089186e+10  2.710286e+11         NaN  

[2160 rows x 11 columns]

In [67]:
################################################### Bitcoin #############################################################
#Creation of a variable for predicting '5' days out into the future
projection_Bitcoin = 5
#creation of a new column with a name prediction
Bitcoin['Prediction'] = Bitcoin[['Close']].shift(-projection_Bitcoin)
Bitcoin

SNo     Name Symbol                 Date          High           Low  \
0        1  Bitcoin    BTC  2013-04-29 23:59:59    147.488007    134.000000   
1        2  Bitcoin    BTC  2013-04-30 23:59:59    146.929993    134.050003   
2        3  Bitcoin    BTC  2013-05-01 23:59:59    139.889999    107.720001   
3        4  Bitcoin    BTC  2013-05-02 23:59:59    125.599998     92.281898   
4        5  Bitcoin    BTC  2013-05-03 23:59:59    108.127998     79.099998   
...    ...      ...    ...                  ...           ...           ...   
2986  2987  Bitcoin    BTC  2021-07-02 23:59:59  33939.588699  32770.680780   
2987  2988  Bitcoin    BTC  2021-07-03 23:59:59  34909.259899  33402.696536   
2988  2989  Bitcoin    BTC  2021-07-04 23:59:59  35937.567147  34396.477458   
2989  2990  Bitcoin    BTC  2021-07-05 23:59:59  35284.344430  33213.661034   
2990  2991  Bitcoin    BTC  2021-07-06 23:59:59  35038.536363  33599.916169   

              Open         Close        Volume     Marketcap  Prediction  
0       134.444000    144.539993  0.000000e+00  1.603769e+09  112.500000  
1       144.000000    139.000000  0.000000e+00  1.542813e+09  115.910004  
2       139.000000    116.989998  0.000000e+00  1.298955e+09  112.300003  
3       116.379997    105.209999  0.000000e+00  1.168517e+09  111.500000  
4       106.250000     97.750000  0.000000e+00  1.085995e+09  113.566002  
...            ...           ...           ...           ...         ...  
2986  33549.600177  33897.048590  3.872897e+10  6.354508e+11         NaN  
2987  33854.421362  34668.548402  2.438396e+10  6.499397e+11         NaN  
2988  34665.564866  35287.779766  2.492431e+10  6.615748e+11         NaN  
2989  35284.344430  33746.002456  2.672155e+10  6.326962e+11         NaN  
2990  33723.509655  34235.193451  2.650126e+10  6.418992e+11         NaN  

[2991 rows x 11 columns]

**DATA VISUALIZATION**

In [68]:
################################################### Monero #############################################################
visualize_Binance_coin = cycle(['Open','Close','High','Low','Prediction'])

fig = px.line(Binance_coin, x=Binance_coin.Date, y=[Binance_coin['Open'], Binance_coin['Close'],
                                        Binance_coin['High'], Binance_coin['Low'], Binance_coin['Prediction']],
             labels={'Date': 'Date','value':'Price'})
fig.update_layout(title_text='Binance_coin', font_size=15, font_color='White',legend_title_text='Parameters')
fig.for_each_trace(lambda t:  t.update(name = next(visualize_Binance_coin)))
fig.update_xaxes(showgrid=False)
fig.update_yaxes(showgrid=False)

fig.show()

In [69]:
################################################### Ethereum #############################################################
visualize_Ethereum = cycle(['Open','Close','High','Low','Prediction'])

fig = px.line(Ethereum, x=Ethereum.Date, y=[Ethereum['Open'], Ethereum['Close'],
                                          Ethereum['High'], Ethereum['Low'],Ethereum['Prediction']],
             labels={'Date': 'Date','value':'Price'})
fig.update_layout(title_text='Ethereum', font_size=15, font_color='White',legend_title_text='Parameters')
fig.for_each_trace(lambda t:  t.update(name = next(visualize_Ethereum)))
fig.update_xaxes(showgrid=False)
fig.update_yaxes(showgrid=False)

fig.show()

In [70]:
################################################### Bitcoin #############################################################
visualize_Bitcoin = cycle(['Open','Close','High','Low','Prediction'])

fig = px.line(Bitcoin, x=Bitcoin.Date, y=[Bitcoin['Open'], Bitcoin['Close'],
                                          Bitcoin['High'], Bitcoin['Low'],Bitcoin['Prediction']],
             labels={'Date': 'Date','value':'Price'})
fig.update_layout(title_text='Bitcoin', font_size=15, font_color='white',legend_title_text='Parameters')
fig.for_each_trace(lambda t:  t.update(name = next(visualize_Bitcoin)))
fig.update_xaxes(showgrid=False)
fig.update_yaxes(showgrid=False)

fig.show()

**CREATION OF THE INDEPENDENT DATA SET(X) & DEPENDENT DATA SET(y)**

################################################### Monero #############################################################

In [71]:
#Creation of the independent data set (X)
X_Binance_coin = np.array(Binance_coin[['Close']])
X_Binance_coin = X_Binance_coin[:-projection_Binance_coin]
print(X_Binance_coin)

[[1.05137996e-01]
 [1.07736997e-01]
 [1.04066998e-01]
 ...
 [3.00211548e+02]
 [3.03295877e+02]
 [2.88218405e+02]]


In [72]:
#creation of the dependent data set (y)
y_Binance_coin = Binance_coin['Prediction'].values
y_Binance_coin = y_Binance_coin[:-projection_Binance_coin]
print(y_Binance_coin)

[1.04249999e-01 9.98668000e-02 1.00451998e-01 ... 3.07732096e+02
 3.02377980e+02 3.20934802e+02]


################################################### Ethereum #############################################################

In [73]:
#Creation of the independent data set (X)
X_Ethereum = np.array(Ethereum[['Close']])
X_Ethereum = X_Ethereum[:-projection_Ethereum]
print(X_Ethereum)

[[7.53324986e-01]
 [7.01897025e-01]
 [7.08447993e-01]
 ...
 [2.16076835e+03]
 [2.27454771e+03]
 [2.11360552e+03]]


In [74]:
#creation of the dependent data set (y)
y_Ethereum = Ethereum['Prediction'].values
y_Ethereum = y_Ethereum[:-projection_Ethereum]
print(y_Ethereum)

[1.82766998e+00 1.82787001e+00 1.68889999e+00 ... 2.32172411e+03
 2.19858246e+03 2.32467945e+03]


################################################### Bitcoin #############################################################

In [75]:
#Creation of the independent data set (X)
X_Bitcoin = np.array(Bitcoin[['Close']])
X_Bitcoin = X_Bitcoin[:-projection_Bitcoin]
print(X_Bitcoin)

[[  144.53999329]
 [  139.        ]
 [  116.98999786]
 ...
 [35867.77773549]
 [35040.83724889]
 [33572.11765253]]


In [76]:
#creation of the dependent data set (y)
y_Bitcoin = Bitcoin['Prediction'].values
y_Bitcoin = y_Bitcoin[:-projection_Bitcoin]
print(y_Bitcoin)

[  112.5          115.91000366   112.30000305 ... 35287.77976616
 33746.00245614 34235.19345116]


**SPLIT THE DATA INTO 85% TRAINING & 15% TESTING DATA SETS**

################################################### Monero #############################################################

In [77]:
x_train_Binance_coin, x_test_Binance_coin, y_train_Binance_coin, y_test_Binance_coin = train_test_split(X_Binance_coin,y_Binance_coin,test_size=0.15)

################################################### Ethereum #############################################################

In [78]:
x_train_Ethereum, x_test_Ethereum, y_train_Ethereum, y_test_Ethereum = train_test_split(X_Ethereum,y_Ethereum,test_size=0.15)

################################################### Bitcoin #############################################################

In [79]:
x_train_Bitcoin, x_test_Bitcoin, y_train_Bitcoin, y_test_Bitcoin = train_test_split(X_Bitcoin,y_Bitcoin,test_size=0.15)

**CREATE & TRAIN THE MODEL : LINEAR REGRESSION**

Linear-regression models have become a proven way to scientifically and reliably predict the future. Because linear regression is a long-established statistical procedure, the properties of linear-regression models are well understood and can be trained very quickly.

################################################### Monero #############################################################

In [80]:
#create & train the model
linReg_Binance_coin = LinearRegression()
linReg_Binance_coin.fit(x_train_Binance_coin,y_train_Binance_coin)

LinearRegression()

################################################### Ethereum #############################################################

In [81]:
linReg_Ethereum = LinearRegression()
linReg_Ethereum.fit(x_train_Ethereum,y_train_Ethereum)

LinearRegression()

################################################### Bitcoin #############################################################

In [82]:
linReg_Bitcoin = LinearRegression()
linReg_Bitcoin.fit(x_train_Bitcoin,y_train_Bitcoin)

LinearRegression()

**EVALUATION**

* Test the model using score which returns the coefficient of determination "R squared" of the prediction.

* R squared coefficient of determination is a statistical measure of how well the regression predictions approximate the real data points.

* In statistics, the coefficient of determination, denoted R2 or r2 and pronounced "R squared", is the proportion of the variation in the dependent variable that is predictable from the independent variable(s).

################################################### Monero #############################################################

In [84]:
linReg_confidence_Binance_coin = linReg_Binance_coin.score(x_test_Binance_coin,y_test_Binance_coin)
print("Linear Regression Confidence for Monero: ",linReg_confidence_Binance_coin)
print(linReg_confidence_Binance_coin*100,'%')

Linear Regression Confidence for Monero:  0.9325821192507862
93.25821192507861 %


################################################### Ethereum #############################################################

In [85]:
linReg_confidence_Ethereum = linReg_Ethereum.score(x_test_Ethereum,y_test_Ethereum)
print("Linear Regression Confidence for Ethereum: ",linReg_confidence_Ethereum)
print(linReg_confidence_Ethereum*100,'%')

Linear Regression Confidence for Ethereum:  0.9776119310413877
97.76119310413877 %


################################################### Bitcoin #############################################################

In [86]:
linReg_confidence_Bitcoin = linReg_Bitcoin.score(x_test_Bitcoin,y_test_Bitcoin)
print("Linear Regression Confidence for Bitcoin: ",linReg_confidence_Bitcoin)
print(linReg_confidence_Bitcoin*100,'%')

Linear Regression Confidence for Bitcoin:  0.98987615954842
98.987615954842 %


**LINEAR REGRESSION MODELS PREDICTION FOR THE NEXT 'N' DAYS**

* Creation of a variable called x_projection_coin & set it equal to the last 5 rows of data from the original dataset.

* Print the linear regression models predictions for the next 5 days(here, N=5).


################################################### Monero #############################################################

In [87]:
x_projection_Binance_coin = np.array(Binance_coin[['Close']])[-projection_Binance_coin:]
print(x_projection_Binance_coin)

[[287.42309381]
 [298.23711708]
 [307.73209557]
 [302.37798013]
 [320.93480178]]


In [88]:
linReg_prediction_Binance_coin = linReg_Binance_coin.predict(x_projection_Binance_coin)
print(linReg_prediction_Binance_coin)

[287.87814201 298.65444736 308.11630859 302.78086761 321.27296547]


################################################### Ethereum #############################################################

In [89]:
x_projection_Ethereum = np.array(Ethereum[['Close']])[-projection_Ethereum:]
print(x_projection_Ethereum)

[[2150.04036404]
 [2226.11428196]
 [2321.72411157]
 [2198.58246387]
 [2324.67944917]]


In [90]:
linReg_prediction_Ethereum = linReg_Ethereum.predict(x_projection_Ethereum)
print(linReg_prediction_Ethereum)

[2140.75109925 2216.24054982 2311.11581988 2188.92025559 2314.04845195]


################################################### Bitcoin #############################################################

In [91]:
x_projection_Bitcoin = np.array(Bitcoin[['Close']])[-projection_Bitcoin:]
print(x_projection_Bitcoin)

[[33897.04858963]
 [34668.54840182]
 [35287.77976616]
 [33746.00245614]
 [34235.19345116]]


In [92]:
linReg_prediction_Bitcoin = linReg_Bitcoin.predict(x_projection_Bitcoin)
print(linReg_prediction_Bitcoin)

[33804.70416239 34572.13916336 35188.10797526 33654.45384765
 34141.06743563]


**INFERENCE**

* Linear Regression Confidence for Monero:  93%

* Linear Regression Confidence for Ethereum:  97%

* Linear Regression Confidence for Bitcoin:  98%


**CONCLUSION**

* Linear regression model's predictions for the next 5 days from the Monero data given here is [287.87814201 298.65444736 308.11630859 302.78086761 321.27296547]

* Linear regression model's predictions for the next 5 days from the Ethereum given here is [2140.75109925 2216.24054982 2311.11581988 2188.92025559 2314.04845195]

* Linear regression model's predictions for the next 5 days from the Bitcoin given here is [34002.65397578 34775.34701974 35395.53611152 33851.37422844
 34341.32182538]



